### PR TITLE
Failing quiz tests

### PR DESCRIPTION
**Pull request by agvidit1**
Adding tests cases for quizzes module which cause it to crash.

PR for Activity #86 #87 #99

### CODE REVIEW

File: test/quizzes_test.py [hunk 1]
```diff
@@ -1,20 +1,63 @@
...
     def setUp(self):
         # Run tests on non-production data
-        self.ctrl = QuizzesController('quizzes_test.py')
-        
+        self.ctrl = QuizzesController('quizzes_test.json')
+
     def test_expose_failure_01(self):
         """
-        Implement this function and two more that
-        execute the code and make it fail.
```
Comment (assiontetteh) [inline test/quizzes_test.py:14]: Test case 01 work fine as expected.

File: test/quizzes_test.py [hunk 2]
```diff
@@ -1,20 +1,63 @@
...
+        # Since there is no quiz, the assertEquals function below fails
+        self.assertEquals(len(quizzes), 1, "There is no quiz.")
         
 
+    def test_expose_failure_02(self):
+        """
+        This test case aims to expose a crash when attempting to add a question to a non-existent id.
+        Expected crash: quizzes_controller.py, line 76
+        """
+        self.ctrl.clear_data()
```
Comment (assiontetteh) [inline test/quizzes_test.py:32]: We need to relook at Test Case 2 as discussed.

File: test/quizzes_test.py [hunk 3]
```diff
@@ -1,20 +1,63 @@
...
+        question_id = self.ctrl.add_question(quiz_id="2333", title="Question 1",text="Sample question")
+        self.assertIsNotNone(question_id,"This question cannot be added")
+
+
+    def test_expose_failure_03(self):
+        """
+        This test case aims to expose a crash when attempting to add an answer with incorrect parameter type.
+        Expected crash: quizzes_controller.py, line 92
+        """
+        self.ctrl.clear_data()
```
Comment (assiontetteh) [inline test/quizzes_test.py:49]: We must also relook at the Test Case 3 as discussed.

**Review by assiontetteh — DISMISSED** [review]
Test Case Case 1, work fine but we need to relook at Test Case 2 and 3 as discussed.

**Review by Saarthak09 — CHANGES_REQUESTED** [review]
In the function test_expose_failure_01 on line 20, instead of passing missing parameter, try passing a variable with a different data type.

File: test/quizzes_test.py [hunk 4]
```diff
@@ -1,20 +1,63 @@
...
-        Implement this function and two more that
-        execute the code and make it fail.
+        This test case aims to expose a crash when adding a quiz with missing parameters.
+        Expected crash: quizzes_controller.py, line 64
         """
-        self.assertTrue(True, 'Example assertion.')
+
+        self.ctrl.clear_data()
+        # Not passing the available_date param
+        # This will give missing positional argument: 'available_date'
```
Comment (Saarthak09) [inline test/quizzes_test.py:19]: This seems correct. Maybe, we can also try to change the variable type instead of passing a missing argument.

**Review by Saarthak09 — DISMISSED** [review]

File: test/quizzes_test.py (hunk 1 above)
Comment (agvidit1) [reply test/quizzes_test.py:14]: I have made a small change to the test case, causing it to fail at quizzes module.

File: test/quizzes_test.py (hunk 2 above)
Comment (agvidit1) [reply test/quizzes_test.py:33]: I have rechecked case 2 and made necessary changes.

File: test/quizzes_test.py (hunk 4 above)
Comment (agvidit1) [reply test/quizzes_test.py]: Notes, I have changes variable type of title.

**Review by Saarthak09 — APPROVED** [review]
@agvidit1 Its looks good. Can be merged.

**Review by Shanks0407 — APPROVED** [review]
@agvidit1 it looks good. The issues raised seem to be resolved, we can go for the merge.

**Review by assiontetteh — COMMENTED** [review]
It can be merged. It looks good.

**Comment by SuDhAnShU-0611** [comment]
It looks good! It can be merged

**Review by SuDhAnShU-0611 — APPROVED** [review]

**Review by assiontetteh — APPROVED** [review]
Look good. Approved.

**Pull request merged**